File: services/dispute/src/main/java/com/example/iter/dispute/domain/repository/DisputeRepository.java
package com.example.iter.dispute.domain.repository;

import com.example.iter.dispute.domain.entity.Dispute;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface DisputeRepository extends JpaRepository<Dispute, Long> {

    Optional<Dispute> findFirstByRentalIdOrderByCreatedAtDesc(Long rentalId);
}

File: services/dispute/src/main/kotlin/com/example/iter/dispute/domain/repository/DisputeRepository.kt
package com.example.iter.dispute.domain.repository

import com.example.iter.dispute.domain.entity.Dispute
import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional

interface DisputeRepository : JpaRepository<Dispute, Long> {

    fun findFirstByRentalIdOrderByCreatedAtDesc(rentalId: Long): Optional<Dispute>
}
